multihost_test/bz_automation/test_automation.py

```diff
@@ -66,3 +66,42 @@ def test_read_faillock_conf_option(self, multihost, create_localusers):
                                "local_anuj --reset")
         execute_cmd(multihost, "cp -vf /etc/security/faillock.conf_bkp "
                                "/etc/security/faillock.conf")
+
+    def test_cve_2010_3316(self, multihost):
+        """
+        :title: CVE-2010-3316-pam_xauth-missing-return-value-checks-from-setuid
+        :id: aebe751c-31b8-11ed-a91f-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=637898
+        """
+        TUSER = "pam-xauth-tester"
+        SLOG = "/var/log/secure"
+        execute_cmd(multihost, "cp -vf /etc/pam.d/su /etc/pam.d/su_anuj")
+        execute_cmd(multihost, "cp -vf /etc/security/limits.conf "
+                               "/etc/security/limits.conf_anuj")
+        execute_cmd(multihost, 'rm -f /tmp/xauthlog')
+        execute_cmd(multihost, "useradd pam-xauth-tester")
+        file_location1 = "/multihost_test/bz_automation/script/myxauth.c"
+        multihost.client[0].transport.put_file(os.getcwd() +
+                                               file_location1,
+                                               '/tmp/myxauth.c')
+        execute_cmd(multihost, "gcc /tmp/myxauth.c -o myxauth")
+        execute_cmd(multihost, "cp -f myxauth /myxauth")
```

Comment:
You could create a function to compile the binary so that you can reuse it another time. Maybe it was already created, I don't know.

Reply:
DOne